### DISCUSSION TITLE
3rd party javascript library destroyed on Custom Field

### DISCUSSION THREAD

**Discussion by diazsasak** [filamentphp/filament#1863] [Help]
### Package

filament/filament

### Package Version

v2.10

### Laravel Version

v8.75

### Livewire Version

v2.10

### PHP Version

PHP 8.0.0

### Bug description

Hi, I have create custom field to edit a view source code.

This is the code:

`
<x-forms::field-wrapper
    :id="$getId()"
    :label="$getLabel()"
    :label-sr-only="$isLabelHidden()"
    :helper-text="$getHelperText()"
    :hint="$getHint()"
    :hint-icon="$getHintIcon()"
    :required="$isRequired()"
    :state-path="$getStatePath()"
>
    
    <textarea
        id="code"
        x-data="
        {
            state: $wire.entangle('{{ $getStatePath() }}')
        }
        "
        x-init="$nextTick(() => {
                CodeMirror.fromTextArea(document.getElementById('code'), {
                    lineNumbers: true,
                    theme: 'dracula',
                    matchBrackets: true,
                    indentUnit: 4,
                    indentWithTabs: true,
                    tabSize: 4,
                    lineWrapping: true,
                })
                 })
        "
        x-text="state"
    ></textarea>
</x-forms::field-wrapper>
`

I saw that on first page load the textarea successfully transformed become codemirror. But after sometime after the other field (SpatieMediaLibraryFileUpload) finish load the image. The textarea is back. Any solution?

Thanks in advance

### Steps to reproduce

![image](https://user-images.githubusercontent.com/7880137/157695333-93c2b161-0bc4-4d50-b96b-9676bffa53f8.png)

![image](https://user-images.githubusercontent.com/7880137/157695428-31baa067-5b40-4579-b970-4751459c7590.png)

**Comment by ryangjchandler** [comment]
@diazsasak Try adding `wire:ignore` to the element with the Alpine logic. This isn't a bug with Filament, so moving to a discussion.

**Reply by diazsasak** [reply]
Hi @ryangjchandler , i have tried using wire:ignore, the textarea is not displayed. it got style="display: none;". 

by the way. this field is not related to table column. when submit, I want to update the view file directly. not saving it to db. this is how it used.

![image](https://user-images.githubusercontent.com/7880137/157778268-2d37a342-456c-4965-a889-46ed2a107954.png)

![image](https://user-images.githubusercontent.com/7880137/157778311-46755cae-7726-44de-ad25-75f284be13fc.png)

is that can make problem?

**Reply by RVxLab** [reply]
I ran into a slightly similar issue when making a package, I solved it by moving all my JS logic out of Alpine. 

Perhaps you can use [my package](https://github.com/RVxLab/filament-colorpicker) to draw inspiration. I’d be happy to help you.
